PWGHF/Tasks/taskMcEfficiency.cxx

```diff
@@ -86,23 +93,163 @@ struct HfTaskMcEfficiency {
+        }
+        if (isHypoMass2TrackStep) {
+          if (pdgCode == pdg::kLambdaCPlus) {
+            massHypo2 = invMassLcToPiKP(candidate);
+          }
+          hCandidates->Fill(kHFStepTracked, pt, massHypo2, pdgCode, cpa, collisionMatched, origin);
+        }
+
+        // check if prongs have passed track cuts
+        if (checkTrack(trackPos) && checkTrack(trackNeg) && checkTrack(trackThird)) {
+          if (isHypoMass1TrackStep) {
+            hCandidates->Fill(kHFStepTrackedCuts, pt, massHypo1, pdgCode, cpa, collisionMatched, origin);
+          }
+          if (isHypoMass2TrackStep) {
+            hCandidates->Fill(kHFStepTrackedCuts, pt, massHypo2, pdgCode, cpa, collisionMatched, origin);
+          }
+        }
+
+        if (!isHypoMass1SelStep && !isHypoMass2SelStep) {
+          continue;
+        }
+
+        // selected candidates
+        if (isHypoMass1SelStep) {
+          hCandidates->Fill(kHFStepTrackedSelected, pt, massHypo1, pdgCode, cpa, collisionMatched, origin);
+        }
+        if (isHypoMass2SelStep) {
+          hCandidates->Fill(kHFStepTrackedSelected, pt, massHypo2, pdgCode, cpa, collisionMatched, origin);
+        }
+
+        // duplicates
+        std::array<int, 3> prongIds = {candidate.prong0Id(), candidate.prong1Id(), candidate.prong2Id()};
+        std::sort(prongIds.begin(), prongIds.end());
+        std::string concat = std::to_string(prongIds[0]) + std::to_string(prongIds[1]) + std::to_string(prongIds[2]);
+        std::size_t hash = std::hash<std::string>{}(concat); /// unique value for the 'concat' string
+        if (duplicates.find(hash) != duplicates.end()) {
+          if (isHypoMass1TrackStep) {
+            hCandidates->Fill(kHFStepTrackedDuplicates, pt, massHypo1, pdgCode, cpa, collisionMatched, origin);
+          }
+          if (isHypoMass2TrackStep) {
+            hCandidates->Fill(kHFStepTrackedDuplicates, pt, massHypo2, pdgCode, cpa, collisionMatched, origin);
+          }
+        }
+        duplicates[hash]++;
+
+      } /// end loop on pdg codes
+    }   /// end loop over candidates
+
+    auto hDuplicateCount = registry.get<TH1>(HIST("hDuplicateCount"));
+    for (const auto& i : duplicates) {
+      hDuplicateCount->Fill(i.second);
+    }
+  }
+
+  template <bool mc, typename T1, typename T2, typename T3>
+  void candidate2ProngLoop(T1& candidates, T2& tracks, T3& mcParticles, std::vector<int> pdgCodes)
   {
     using TracksType = std::decay_t<decltype(tracks)>;
 
     auto hCandidates = registry.get<StepTHn>(HIST("hCandidates"));
     std::map<int64_t, int> duplicates;
 
-    for (const auto pdgCode : pdgCodes.value) {
+    for (const auto pdgCode : pdgCodes) {
       auto decayType = -1;
       std::array<int, 2> pdgDaughters;
 
       if (pdgCode == pdg::kD0) {
-        decayType = 1 << DecayType::D0ToPiK;
+        decayType = 1 << o2::aod::hf_cand_2prong::DecayType::D0ToPiK;
```

Comment:
```suggestion
        decayType = 1 << aod::hf_cand_2prong::DecayType::D0ToPiK;
```

Reply:
And everywhere else we can skip the `o2::` as we are using the namespace

Reply:
Done, thanks